sdks/java/extensions/google-cloud-platform-core/src/main/java/org/apache/beam/sdk/extensions/gcp/util/GcsUtil.java

```diff
@@ -739,7 +741,42 @@ public WritableByteChannel create(GcsPath path, CreateOptions options) throws IO
 
   GoogleCloudStorage createGoogleCloudStorage(
       GoogleCloudStorageOptions options, Storage storage, Credentials credentials) {
-    return new GoogleCloudStorageImpl(options, storage, credentials);
+    try {
+      // Attempt to construct gcs-connector 3.x-style GoogleCloudStorage, which is created
+      // exclusively via Builder method; this can be replaced once Java 8 is dropped and
```

Comment:
could add a // TODO tag like

// TODO eliminate reflection once Beam drops Java 8 support and upgrade to use gcsio 3.x

Reply:
good idea- updated 👍